account_invoice_report_hls/report/invoice_delivery_report_line.py

```diff
@@ -7,36 +7,31 @@
 from odoo.tools.float_utils import float_compare, float_repr
 
 
-class InvoiceDeliveryReport(models.TransientModel):
-    _name = "invoice.delivery.report"
+class AccountInvoice(models.Model):
+    _inherit = "account.invoice"
 
-    invoice_id = fields.Many2one("account.invoice")
-    line_ids = fields.One2many("invoice.delivery.report.line", inverse_name="report_id")
+    report_line_ids = fields.One2many(
+        "invoice.delivery.report.line", inverse_name="report_id"
+    )
 
     @api.model
-    def _create_invoice_delivery_report(self, invoices):
-        invoice_reports = self.browse()
-        for invoice in invoices:
-            report = self.create({"invoice_id": invoice.id})
+    def _create_invoice_delivery_report(self):
+        for invoice in self:
+            # if we use unlink() here,
+            # action_invoice_sent method behavior will be broken
+            invoice.write({"report_line_ids": [(5, 0, 0)]})
             self.env[
                 "invoice.delivery.report.line"
             ]._create_invoice_delivery_report_lines(
-                report, invoice.invoice_line_ids, invoice.date_from, invoice.date_to
+                invoice, invoice.invoice_line_ids, invoice.date_from, invoice.date_to
             )
-            invoice_reports += report
-        return invoice_reports
-
-    @api.multi
-    def _get_report_base_filename(self):
-        self.ensure_one()
-        return self.invoice_id._get_report_base_filename()
 
 
 class InvoiceDeliveryReportLine(models.TransientModel):
     _name = "invoice.delivery.report.line"
     _order = "date_delivered"
 
-    report_id = fields.Many2one("invoice.delivery.report")
+    report_id = fields.Many2one("account.invoice")
```

Comment:
Would make more sense to change the field name.
```suggestion
    invoice_id = fields.Many2one("account.invoice")
```